File: clients/python/examples/remote_cockpit.py
#!/usr/bin/env python3
from __future__ import annotations

import os
import signal
import time

from glass_client import GlassClient

stopped = False


def stop(*_: object) -> None:
    global stopped
    stopped = True


signal.signal(signal.SIGINT, stop)
glass = GlassClient(command=os.environ.get("GLASS_BINARY", "glass"))
try:
    trust = glass.call("glass.workspace.trust.status")
    runtime = glass.call("glass.runtime.inspect")
    browser = glass.call("glass.browser.state")
    project = runtime.get("project") if isinstance(runtime, dict) else {}
    print(
        {
            "trust": trust.get("trust") if isinstance(trust, dict) else trust,
            "root": project.get("root") if isinstance(project, dict) else None,
            "browserConnected": browser.get("connected") if isinstance(browser, dict) else None,
        }
    )
    while not stopped:
        print(glass.call("glass.workspace.trust.inspect"))
        time.sleep(2)
except KeyboardInterrupt:
    pass
finally:
    glass.close()

File: clients/typescript/examples/remote-cockpit.ts
import { GlassClient } from "../src/index.ts";

const glass = new GlassClient({ command: process.env.GLASS_BINARY ?? "glass" });
const controller = new AbortController();
process.once("SIGINT", () => controller.abort());

try {
  const trust = await glass.call("glass.workspace.trust.status") as { trust?: string };
  const runtime = await glass.call("glass.runtime.inspect") as { project?: { root?: string } };
  const browser = await glass.call("glass.browser.state") as { connected?: boolean };
  console.log({
    trust: trust.trust,
    root: runtime.project?.root,
    browserConnected: browser.connected,
  });
  while (!controller.signal.aborted) {
    console.log(await glass.call("glass.workspace.trust.inspect"));
    await new Promise((resolve) => {
      const timer = setTimeout(resolve, 2_000);
      controller.signal.addEventListener("abort", () => {
        clearTimeout(timer);
        resolve(undefined);
      }, { once: true });
    });
  }
} finally {
  glass.close();
}
